### PR TITLE
Fix simple spot lights for mobile

### PR DESCRIPTION
**Pull request by moudgils**
## What does this PR do?

The alignment one the cpu and gpu was incorrect causing a mismatch related to spot light data

## How was this PR tested?

Tested on iOS

### CODE REVIEW

**Comment by akioCL** [comment]
Why wasn't this caught by the validation on the RHI? I think we check if the alignment are off between GPU and CPU

File: Gems/Atom/Feature/Common/Assets/ShaderResourceGroups/CoreLights/ViewSrg.azsli [hunk 1]
```diff
@@ -44,6 +44,7 @@ partial ShaderResourceGroup ViewSrg
         float m_affectsGIFactor;
         bool m_affectsGI;
         float3 m_padding;
```
Comment (akioCL) [inline Gems/Atom/Feature/Common/Assets/ShaderResourceGroups/CoreLights/ViewSrg.azsli:46]: Do we still need this padding if we have [[pad_to(16)]]?

Comment (moudgils) [reply Gems/Atom/Feature/Common/Assets/ShaderResourceGroups/CoreLights/ViewSrg.azsli:46]: In theory, no. But I wanted to keep it in order to match the padding in C++.

**Comment by moudgils** [comment]
> Why wasn't this caught by the validation on the RHI? I think we check if the alignment are off between GPU and CPU

RHI does validation from Reflection stride numbers from shader pipeline vs runtime stride. In this case it was same. But spirv-cross I think was padding the struct under the hood hence resulting in the mismatch between gpu and cpu. My fix ensures the stride is 16 bytes on both sides so that spirv-cross doesn't do any extra padding on top. Plus in this case we were wasting space on an unneeded padding float.

**Review by AMZN-Gene — APPROVED** [review]

**Review by akioCL — APPROVED** [review]

**Pull request merged**